admin_app/src/utils/api.ts

```diff
@@ -156,296 +127,223 @@ const bulkUploadContents = async (file: File, token: string) => {
-    } else {
-      throw new Error("Error deleting urgency rule");
-    }
-  });
+  try {
+    const response = await api.delete(`/urgency-rules/${rule_id}`, {
+      headers: { Authorization: `Bearer ${token}` },
+    });
+    return response.data;
+  } catch (error) {
+    throw new Error("Error deleting urgency rule");
+  }
 };
 
 const getLoginToken = async (username: string, password: string) => {
   const formData = new FormData();
   formData.append("username", username);
   formData.append("password", password);
-  const response = await fetch(`${NEXT_PUBLIC_BACKEND_URL}/login`, {
-    method: "POST",
-    body: formData,
-  });
 
-  if (response.ok) {
-    let response_json = await response.json();
-    return response_json;
-  } else {
-    console.log("Error fetching login token", response);
-    throw response;
+  try {
+    const response = await api.post("/login", formData, {
+      headers: {
+        "Content-Type": "multipart/form-data",
+      },
+    });
+    return response.data;
+  } catch (error) {
+    console.log(error);
+    throw new Error("Error fetching login token");
   }
 };
 
 const getGoogleLoginToken = async (idToken: {
   client_id: string;
   credential: string;
 }) => {
-  return fetch(`${NEXT_PUBLIC_BACKEND_URL}/login-google`, {
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-    },
-    body: JSON.stringify(idToken),
-  }).then((response) => {
-    if (response.ok) {
-      let resp = response.json();
-      return resp;
-    } else {
-      throw new Error("Error fetching login token");
-    }
-  });
+  try {
+    const response = await api.post("/login-google", idToken, {
+      headers: { "Content-Type": "application/json" },
+    });
+    return response.data;
+  } catch (error) {
+    throw new Error("Error fetching Google login token");
+  }
 };
 
+// Perform search
 const getSearch = async (
   question: string,
   generate_llm_response: boolean,
   token: string,
 ) => {
-  const embeddingUrl = `${NEXT_PUBLIC_BACKEND_URL}/search`;
-  return fetch(embeddingUrl, {
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-      Authorization: `Bearer ${token}`,
-    },
-    body: JSON.stringify({
-      query_text: question,
-      generate_llm_response: generate_llm_response,
-    }),
-  })
-    .then((response) => {
-      return response.json().then((data) => {
-        const responseWithStatus = { status: response.status, ...data };
-        return responseWithStatus;
-      });
-    })
-    .catch((error) => {
-      console.error("Error:", error);
-      throw error;
-    });
+  try {
+    const response = await api.post(
+      "/search",
+      {
+        query_text: question,
+        generate_llm_response,
+      },
+      {
+        headers: { Authorization: `Bearer ${token}` },
+      },
+    );
+    return { status: response.status, ...response.data };
+  } catch (error) {
+    throw new Error("Error performing search");
+  }
```

Comment:
I can't test Search properly because my dev environment has issues with the pytorch or smt but I think this section needs to handle errors slightly differently to be compatible with the `processNotOKResponse` of the SideBars... Or maybe `processNotOKResponse` needs to change. We want to show the error code and let the user see the response in the JSON

Reply:
I will check it out.

Reply:
Fixed it